src/reconcile/rec_col.c

```diff
@@ -1171,7 +1171,7 @@ __rec_col_var_helper(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_SALVAGE_COOKI
         WT_RET(__wt_rec_split_crossing_bnd(session, r, val->len));
 
     /* Copy the value onto the page. */
-    if (!deleted && ovfl_usedp == NULL && btree->dictionary)
+    if (dictionary && btree->dictionary && !deleted && ovfl_usedp == NULL)
```

Comment:
This is confusing - I asked "why are there two different dictionary booleans"? And tried to go and figure out the answer. It would be better if the comment for the conditional made it clear how the parameter and btree `dictionary` variables differ. Naming them differently would be good too - if you can find a better name for the parameter.

Reply:
I was confused too the first time I read that code - having two people feeling the same is enough to do something about it, thank you for raising it. For future readers:

- `btree->dictionary` indicates whether the dictionary feature is enabled for that btree
- `dictionary` indicates whether we should use a dictionary for that specific value. If `btree->dictionary` is disabled, this one does not really matter.

I will try to come up with better names.

Reply:
> I'm not the right person to review this change.

@quchenhao, would you be ok to review these changes?